main.go

```diff
@@ -0,0 +1,44 @@
+/*
+ *Author: Stefan
+ *Date: 11/16/2019
+ *Task: Implement the following methods: End - defines the end of the shape, Perimiter, Area
+**/
+
+package main
+
+import "fmt"
+
+//Point structure
+type Point struct {
+	x, y int
+}
+
+//Square structure
+type Square struct {
+	start Point
+	a     uint
+}
+
+//End method
+func (p *Square) End() (int, int) {
+	var xend = p.start.x + int(p.a)
+	var yend = p.start.y - int(p.a)
+	return xend, yend
+}
+
+//Perimeter method
+func (p *Square) Perimeter() uint {
+	return p.a * uint(4)
```

Comment:
``` return p.a * 4```

Reply:
When I wrote this way, the compiler issued a warning. Since there is no automatic conversion, and p.a is of type uint, and 4 is of type int.